### PR TITLE
Introducing pesel validation

### PR DESCRIPTION
**Pull request by Aleksiiej**
PESEL validation method has been added into Person class
image

Also some cleanup in test structure has been performed

### CODE REVIEW

File: src/app/person/person.cpp [hunk 1]
```diff
@@ -32,11 +35,78 @@ std::string Person::getSurname() const
 
 void Person::setPesel(const std::string& pesel)
 {
-    // TODO Validation
-    pesel_ = pesel;
+    if (validatePESEL(pesel))
+    {
+        pesel_ = pesel;
+    }
+    // TODO: Else what?
```
Comment (fbtom) [inline src/app/person/person.cpp:42]: Instead if condition maybe throw some exception.

File: src/app/person/person.cpp [hunk 2]
```diff
@@ -32,11 +35,78 @@ std::string Person::getSurname() const
...
 }
 
 std::string Person::getPesel() const
 {
     return pesel_;
 }
+
+bool Person::validatePESEL(const std::string& PESEL) const
```
Comment (fbtom) [inline src/app/person/person.cpp:50]: Please swap `PESEL` to `Pesel`
and argument in lowercase

File: src/app/person/person.cpp [hunk 3]
```diff
@@ -32,11 +35,78 @@ std::string Person::getSurname() const
...
+    if (PESEL.size() != 11)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool Person::validatePESELDate(const std::string& PESEL) const
+{
+    std::string Month = PESEL.substr(2, 2);
```
Comment (fbtom) [inline src/app/person/person.cpp:70]: lowercase for variables

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Review by BaqWin — COMMENTED** [review]
Consider adding Gender enum for better pesel validation

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

File: src/app/person/person.cpp (hunk 1 above)
Comment (Aleksiiej) [reply src/app/person/person.cpp:42]: Condition has been removed, currently there is no pesel checking in function setPesel()

File: src/app/person/person.cpp (hunk 3 above)
Comment (Aleksiiej) [reply src/app/person/person.cpp:70]: Done

File: src/app/person/person.cpp [hunk 4]
```diff
@@ -32,11 +35,74 @@ std::string Person::getSurname() const
...
 }
 
 std::string Person::getPesel() const
 {
     return pesel_;
 }
+
+bool Person::validatePesel(const std::string& pesel) const
+{
+    if (/*not validatePeselSize(pesel) or not validatePeselDate(pesel) or*/ not validateControlNumber(pesel))
```
Comment (fbtom) [inline src/app/person/person.cpp:48]: Why it is commented out? If methods are not necessary please then remove them

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: src/app/person/person.cpp (hunk 4 above)
Comment (Aleksiiej) [reply src/app/person/person.cpp]: Done

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Review by BaqWin — APPROVED** [review]

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @Aleksiiej for making course materials better!

**Pull request merged**